src/mol-model/structure/structure/mapping.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2024 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Alexander Rose <alexander.rose@weirdbyte.de>
+ */
+
+import { OrderedSet } from '../../../mol-data/int/ordered-set';
+import { ElementIndex } from '../model/indexing';
+import { Structure } from './structure';
+import { Unit } from './unit';
+
+/** Serial index of an element in the structure across all units */
+export type SerialIndex = { readonly '@type': 'serial-index' } & number
+
+export interface SerialMapping {
+    /** Cumulative count of preceding elements for each unit */
+    cumulativeUnitElementCount: ArrayLike<number>
+    /** Unit index for each serial element in the structure */
+    unitIndices: ArrayLike<number>
+    /** Element index for each serial element in the structure */
+    elementIndices: ArrayLike<ElementIndex>
```

Comment:
Do we need this? The both unit and element indices can be obtained from a linear index and the `cumulativeUnitElementCount` by doing a `findPredecessorIndex` and subtracting to get `UnitIndex`.

Even if we want to skip the `findPredecessorIndex` (since the log access can have some downsides which I am not aware of now), the `elementIndices` are still reduntant for the same reason (subtraction gets you the `UnitIndex` from which you can get the `ElementIndex`). That's a lot of memory saved for large structures.